patroni/postgresql.py

```diff
@@ -126,7 +126,12 @@ def get_local_address(self):
         return local_address + ':' + self.port
 
     def get_postgres_role_from_data_directory(self):
```

Comment:
This function is called from the set_role, so it should return only those roles accepted by set_role (which is not 'uninitialized').